aspnetcore/migration/inc/abtesting.md

```diff
@@ -0,0 +1,146 @@
+---
+title: A/B Testing Migrated Endpoints
+description: A/B Testing Migrated Endpoints
+author: twsouthwick
+ms.author: tasou
+monikerRange: '>= aspnetcore-6.0'
+ms.date: 3/1/2023
+ms.topic: article
+ms.prod: aspnet-core
+uid: migration/inc/ab-testing
+---
+
+# A/B Testing Endpoints during Migration
+
+During incremental migration, new endpoints will be brought over to a YARP-enabled ASP.NET Core application. With the default setup, these will automatically be served for all requests once deployed. In order to test these endpoints, or be able to turn them off if needed, we need to add some additional setup for the application.
+
+This document will describe how to setup a conditional endpoint selection system to enable A/B testing during incremental migration. It assumes a setup as described [incremental migration overview](xref:migration/inc/overview) as a starting point.
+
+## Conditional Endpoint Selection
+
+To enable conditional endpoint selection, a few services need to be defined:
+
+1. Metadata that can be added to an endpoint (controllers, minimal APIs, etc) to turn on any conditional related logic. If this is not present on an endpoint, that endpoint will not incur any potential cost with checks that may be performed.
+
+    ```CSharp
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public sealed class ConditionalRouteAttribute : Attribute
+    {
+        public bool IsConditional { get; set; }
```

Comment:
Should this default to true? Any routes without this are assumed unconditional.

Reply:
Yeah, that's a better default. Updated